### PR TITLE
chore!: ACVM 0.24

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

## Problem\*

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
Seems we have a circular dependency due to the backend being in here when we are making serialization changes:

1. This PR requires a new version of barretenberg to pass
2. The barretenberg PR requires* this branch to be merged into master to pass

*We could merge barretenberg while tests link to this branch, then merge this branch, then change the tests on bb to point to master again. 

We could also push changes to AztecProtocol/barretenberg, push a nightly version, update this to point to the nightly, merge this, then merge barretenberg once this is in master, then change this to point to the released barretenberg

**Comment by kevaundray** [comment]
I'll open up a PR so that acvm-backend-barretenberg references the nargo_cli tests

File: Cargo.toml [hunk 1]
```diff
@@ -28,7 +28,7 @@ edition = "2021"
 rust-version = "1.66"
 
 [workspace.dependencies]
-acvm = "0.23.0"
+acvm = { git = "https://github.com/noir-lang/acvm", branch = "kw/remove-btreemap-temp" }
```
Comment (sirasistant) [inline Cargo.toml:31]:
```suggestion
acvm = "0.24.0"
```
:D

Comment (kevaundray) [reply Cargo.toml]: Gracias ser :)

File: crates/nargo/Cargo.toml [hunk 2]
```diff
@@ -11,6 +11,10 @@ edition.workspace = true
 rustc_version = "0.4.0"
 
 [dependencies]
+
+# Backends
+acvm-backend-barretenberg = { path = "../acvm_backend_barretenberg" }
```
Comment (TomAFrench) [inline crates/nargo/Cargo.toml:16]: nargo core is meant to be independent of its environment so linking it to the CLI like this isn't correct. We should remove the prove/verify operations from this package in the meantime.

Comment (kevaundray) [reply crates/nargo/Cargo.toml]: yep good point

**Review by sirasistant — APPROVED** [review]
LGTM

**Pull request merged**